### PR TITLE
Rebuild with `hdf5` version `1.14.3`

### PR DESCRIPTION
**Pull request by apmorton**
xref: https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/611#issuecomment-1852946346

<hr>

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: .ci_support/osx_arm64_mpiopenmpinumpy1.26python3.12.____cpython.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ channel_sources:
 channel_targets:
 - conda-forge main
 hdf5:
-- 1.14.2
+- 1.14.3
```
Comment (hmaarrfk) [inline .ci_support/osx_arm64_mpiopenmpinumpy1.26python3.12.____cpython.yaml:12]: did you manually change the pin? I'm mostly curious as to how the migration works.

Was rerendering enough to get this pin to take effect?

**Comment by jakirkham** [comment]
Tried to change the title and description a bit to make this a little clearer. Though please add a bit more to the description (as others may come to this PR trying to understand what is changed or why)

File: .ci_support/osx_arm64_mpiopenmpinumpy1.26python3.12.____cpython.yaml (hunk 1 above)
Comment (apmorton) [reply .ci_support/osx_arm64_mpiopenmpinumpy1.26python3.12.____cpython.yaml:12]: I did nothing besides rerender and bump the build number.

Comment (hmaarrfk) [reply .ci_support/osx_arm64_mpiopenmpinumpy1.26python3.12.____cpython.yaml:12]: great. thanks. I was hoping this would be the behavior

**Comment by hmaarrfk** [comment]
I restarted the failed build and it seemed to have fixed things

**Comment by hmaarrfk** [comment]
Thank you!

**Pull request merged**